compiler/noirc_evaluator/src/ssa/interpreter/errors.rs

```diff
@@ -0,0 +1,105 @@
+    #[error("Block {block} is missing the terminator instruction")]
+    BlockMissingTerminator { block: BasicBlockId },
+    #[error("constrain {lhs_id} == {rhs_id} failed:\n    {lhs} != {rhs}")]
+    ConstrainEqFailed { lhs: String, lhs_id: ValueId, rhs: String, rhs_id: ValueId },
+    #[error("constrain {lhs_id} != {rhs_id} failed:\n    {lhs} == {rhs}")]
+    ConstrainNeFailed { lhs: String, lhs_id: ValueId, rhs: String, rhs_id: ValueId },
+    #[error(
+        "Range check of {value_id} = {value} failed.\n  Max bits allowed by range check = {max_bits}\n  Actual bit count = {actual_bits}"
+    )]
+    RangeCheckFailed { value: String, value_id: ValueId, actual_bits: u32, max_bits: u32 },
+    #[error(
+        "Range check of {value_id} = {value} failed.\n  Max bits allowed by range check = {max_bits}\n  Actual bit count = {actual_bits}\n  {message}"
+    )]
+    RangeCheckFailedWithMessage {
+        value: String,
+        value_id: ValueId,
+        actual_bits: u32,
+        max_bits: u32,
+        message: String,
+    },
+    #[error("Call to unknown foreign function {name}")]
+    UnknownForeignFunctionCall { name: String },
+    #[error("Cannot call non-function value {value_id} = {value}")]
+    CalledNonFunction { value: String, value_id: ValueId },
+    // Note that we don't need to display the value_id because displaying a reference
+    // value shows the original value id anyway
+    #[error("Reference value `{value}` passed from a constrained to an unconstrained function")]
+    ReferenceValueCrossedUnconstrainedBoundary { value: String },
+    #[error("Reference value `{value}` loaded before it was first stored to")]
+    UninitializedReferenceValueLoaded { value: String },
+    #[error(
+        "Mismatched types in binary operator: `{operator} {lhs_id}, {rhs_id}`  ({operator} {lhs}, {rhs})"
+    )]
+    MismatchedTypesInBinaryOperator {
+        lhs_id: ValueId,
+        lhs: String,
+        operator: BinaryOp,
+        rhs_id: ValueId,
+        rhs: String,
+    },
+    #[error("Division by zero: `div {lhs_id}, {rhs_id}`  ({lhs} / {rhs})")]
+    DivisionByZero { lhs_id: ValueId, lhs: String, rhs_id: ValueId, rhs: String },
+    #[error("Unsupported operator `{operator}` for type `{typ}`")]
+    UnsupportedOperatorForType { operator: &'static str, typ: &'static str },
+    #[error(
+        "Invalid bit size of `{bit_size}` given to truncate, maximum size allowed for unsigned values is {MAX_UNSIGNED_BIT_SIZE}"
+    )]
+    InvalidUnsignedTruncateBitSize { bit_size: u32 },
+    #[error(
+        "Invalid bit size of `{bit_size}` given to truncate, maximum size allowed for signed values is {MAX_SIGNED_BIT_SIZE}"
+    )]
+    InvalidSignedTruncateBitSize { bit_size: u32 },
+    #[error("Rhs of `{operator}` should be a u32 but found `{rhs_id} = {rhs}`")]
+    RhsOfBitShiftShouldBeU32 { operator: &'static str, rhs_id: ValueId, rhs: String },
```

Comment:
Hm, I thought it had to be u8 🤔

Reply:
For example the SSA of this:
```rust
fn main(a: u64) -> pub u64 {
    a << 13
}
```

shows it's `u8`:

```
After Initial SSA (1):
acir(inline) fn main f0 {
  b0(v0: u64):
    v2 = shl v0, u8 13
    v3 = truncate v2 to 64 bits, max_bit_size: 65
    return v2
}
```

Reply:
Yes, this should be a u8

Reply:
fixed